pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -848,19 +848,21 @@ func TestHttpsAppProbe(t *testing.T) {
 		},
 	}
 	for _, tc := range testCases {
-		client := http.Client{}
-		req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost%s", tc.probePath), nil)
-		if err != nil {
-			t.Errorf("[%v] failed to create request", tc.probePath)
-		}
-		resp, err := client.Do(req)
-		if err != nil {
-			t.Fatal("request failed")
-		}
-		defer resp.Body.Close()
-		if resp.StatusCode != tc.statusCode {
-			t.Errorf("[%v] unexpected status code, want = %v, got = %v", tc.probePath, tc.statusCode, resp.StatusCode)
-		}
+		func() {
```

Comment:
use t.Run(), not `func()` https://dave.cheney.net/2019/05/07/prefer-table-driven-tests